File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.ts
class CustomStack {
    private stk: number[];
    private add: number[];
    private i: number;

    constructor(maxSize: number) {
        this.stk = Array(maxSize).fill(0);
        this.add = Array(maxSize).fill(0);
        this.i = 0;
    }

    push(x: number): void {
        if (this.i < this.stk.length) {
            this.stk[this.i++] = x;
        }
    }

    pop(): number {
        if (this.i <= 0) {
            return -1;
        }
        let val = this.stk[--this.i];
        this.stk[this.i] = 0;
        return val;
    }

    increment(k: number, val: number): void {
        let idx = 0;
        while (idx < k) {
            this.stk[idx++]+= val;
        }
    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * var obj = new CustomStack(maxSize)
 * obj.push(x)
 * var param_2 = obj.pop()
 * obj.increment(k,val)
 */

File: 1381-design-a-stack-with-increment-operation/1381-design-a-stack-with-increment-operation.java
class CustomStack {
    int mSize = 0;
    Stack<Integer> s = new Stack<Integer>();
    
    public CustomStack(int maxSize) {
        mSize = maxSize;
        s = new Stack<Integer>();
    }
    
    public void push(int x) {
        if(s.size() < mSize){
            s.push(x);
        }
    }
    
    public int pop() {
        if(s.size() == 0){
            return -1;
        }
       return s.pop();
    }
    
    public void increment(int k, int val) {
        Stack<Integer> tmp  = new Stack<Integer>();
        while(!s.empty()){
            if(s.size()<=k){
                tmp.push(s.pop()+val);
            }else{
                tmp.push(s.pop());
            }
        }
        while(!tmp.empty()){
            s.push(tmp.pop());
        }
    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack obj = new CustomStack(maxSize);
 * obj.push(x);
 * int param_2 = obj.pop();
 * obj.increment(k,val);
 */